dumpling/export/config.go

```diff
@@ -313,6 +342,7 @@ func (*Config) DefineFlags(flags *pflag.FlagSet) {
 	flags.Bool(flagTransactionalConsistency, true, "Only support transactional consistency")
 	_ = flags.MarkHidden(flagTransactionalConsistency)
 	flags.StringP(flagCompress, "c", "", "Compress output file type, support 'gzip', 'snappy', 'zstd', 'no-compression' now")
+	flags.String(flagOutputDialect, "default", "The dialect of output file, support 'default', 'snowflake', 'redshift' now")
```

Comment:
Maybe it better to add a "binary-encoding" option with available value of "quote-escaped", "base64", "hex", "hex-with-leading-0x".

If we use "snowflake", "redshift" as an user I think this software will promise to keep all feature capable with "snowflake", "redshift"

Reply:
Yes, you are right. The purpose of `output-dialect` is exactly what you mentioned: We would like the output of our toolchain to be compatible with Redshift / Snowflake.

https://github.com/pingcap/tidb/issues/47964

If we only introduce binary-encoding without compatibility promising, then this PR may not resolve the above issue, after all it will only introduce a new feature called "transform binary encoding" instead of "allow the CSV being ingested by Redshift".